lighthouse-core/scripts/pptr-run-devtools.js

```diff
@@ -152,122 +183,147 @@ const startLighthouse = `
       }
     }
   }
+}
+
+async function runLighthouse() {
+  // @ts-expect-error global
+  const panel = UI.panels.lighthouse || UI.panels.audits;
+
+  /** @type {Promise<{lhr: LH.Result, artifacts: LH.Artifacts}>} */
+  const resultPromise = new Promise((resolve, reject) => {
+    const methodName = panel.__proto__.buildReportUI ? 'buildReportUI' : '_buildReportUI';
+    addSniffer(
+      panel.__proto__,
+      methodName,
+      /**
+       * @param {LH.Result} lhr
+       * @param {LH.Artifacts} artifacts
+       */
+      (lhr, artifacts) => resolve({lhr, artifacts})
+    );
+
+    addSniffer(
+      panel.statusView.__proto__,
+      'renderBugReport',
+      reject,
+    );
+  });
 
+  const button = panel.contentElement.querySelector('button');
   button.click();
```

Comment:
```suggestion
  await button.click();
```

Reply:
?

https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/click

This isn't a JS handle, is executed in the page

Reply:
oh i see